packages/utilities/src/BaseComponent.ts

```diff
@@ -2,20 +2,14 @@ import * as React from 'react';
 import { Async } from './Async';
 import { EventGroup } from './EventGroup';
 import { IDisposable } from './IDisposable';
-import { warnDeprecations, warnMutuallyExclusive, warnConditionallyRequiredProps, ISettingsMap } from './warn';
+import { ISettingsMap } from './warn/warn';
+import { warnConditionallyRequiredProps } from './warn/warnConditionallyRequiredProps';
+import { warnMutuallyExclusive } from './warn/warnMutuallyExclusive';
+import { warnDeprecations } from './warn/warnDeprecations';
 import { initializeFocusRects } from './initializeFocusRects';
 import { initializeDir } from './initializeDir';
 import { IRefObject } from './createRef';
-
-/**
- * BaseProps interface.
- *
- * @public
- */
-// tslint:disable-next-line:no-any
-export interface IBaseProps<T = any> {
-  componentRef?: IRefObject<T>;
-}
+import { IBaseProps } from './BaseComponent.types';
```

Comment:
Separating this out means that code which relies on the props only can import the props without BaseComponent sneaking into the bundle treemap.